main.tf

```diff
@@ -341,6 +291,80 @@ resource "checkly_alert_channel" "pagerduty_ac" {
 resource "checkly_alert_channel" "slack_ac" {
   slack {
     channel = "#checkly-notifications"
-    url = "https://slack.com/webhook"
+    url     = "https://hooks.slack.com/services/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g"
+  }
+}
+
+
+resource "checkly_check_group" "test_group1" {
+  name      = "Critical User Flows"
+  activated = true
+  muted     = false
+  tags = [
+    "terraform", "critical-user-flows"
+  ]
+  concurrency = 300
+  locations = [
+    "us-west-1",
+  ]
+}
+
+resource "checkly_check" "browser-check" {
+  for_each = fileset("${path.module}/critical-user-flows", "*") // Iterates through the files in the scripts folder in your project's directory
+
+  name         = each.key // Sets the check name to match the file's 
+  group_id     = checkly_check_group.test_group1.id
+  type         = "BROWSER"
+  activated    = true
+  frequency    = 60
+  double_check = true
+  ssl_check    = false
+  locations = [
+    "us-west-1",
+    "us-east-1"
+  ]
+  use_global_alert_settings = false
+
+  alert_channel_subscription {
+    channel_id = checkly_alert_channel.slack_ac.id
+    activated  = true
+  }
+  environment_variables = { // This sets check-level environment variables
+    TEST_EMAIL      = "user@email.com",
+    TEST_USER_ID    = "7d8d8288-43a6-44b0-ba08-4c5736c1b6ed",
+    USER_EMAIL      = "user@email.com",
+    USER_PASSWORD   = "supersecure1",
+    FILE_PATH       = "tests/fixtures/file.jpg",
+    TEST_FILE_PATH  = "tests/fixtures/invoice-20-07-02.pdf",
+    ENVIRONMENT_URL = "https://checklyhq.com"
   }
-}
\ No newline at end of file
+
+  script = file("${path.module}/critical-user-flows/${each.key}") // Assigns the script contained in each file to each new created check resource
+}
+
+resource "checkly_maintenance_windows" "maintenance-1" {
+  name            = "Maintenance Windows"
+  starts_at       = "2022-04-24T00:00:00.000Z"
+  ends_at         = "2022-04-25T00:00:00.000Z"
+  repeat_unit     = "MONTH"
+  repeat_ends_at  = "2023-08-24T00:00:00.000Z"
+  repeat_interval = 1
+  tags = [
+    "auto", "terraform"
+  ]
+}
+
+resource "checkly_dashboard" "dashboard-main" {
+  custom_url      = "danube123"                            // A unique subdomain name under "checklyhq.com"
```

Comment:
@modern-sapien just making sure: did you test if the `apply` fails in cases where the dashboard subdomain already exists? My rationale is that the first try after the repo clone should be successful for good UX.